04.ls/ls.rb

```diff
@@ -3,44 +3,45 @@
 
 require 'optparse'
 
-NUMBER_OF_COL_MAX = 3
+MAX_COL_COUNT = 3
 SPACE_WIDTH = 2
 
-def search_option(argv)
+def parse_options(argv)
   opt = OptionParser.new
   options = {}
   opt.on('-a') { |v| options[:a] = v }
-  file_paths = opt.parse(argv)
-  [options, file_paths]
+  directory_paths = opt.parse(argv)
+  [options, directory_paths]
 end
 
-def get_filenames(target_path, options)
+def fetch_file_names(target_directory_path, options)
   dotmatch_flag = options[:a] ? File::FNM_DOTMATCH : 0
-  Dir.glob('*', dotmatch_flag, base: target_path)
+  Dir.glob('*', dotmatch_flag, base: target_directory_path)
 end
 
-def output(filenames)
-  return if filenames.empty?
+def output(file_names)
+  return if file_names.empty?
 
-  number_of_row = ((filenames.size - 1) / NUMBER_OF_COL_MAX) + 1
-  number_of_col = [filenames.size, NUMBER_OF_COL_MAX].min
+  row_count = ((file_names.size - 1) / MAX_COL_COUNT) + 1
+  col_count = [file_names.size, MAX_COL_COUNT].min
 
   # NOTE: OS標準のlsコマンドは横並びではなく縦並びで出力される(転置して出力される)
-  # NOTE: filenames_tableの要素は行と列が出力したい形(縦並び)とは逆で保存されている
-  filenames_table = filenames.each_slice(number_of_row).to_a
-  widths = filenames_table.map { |col| col.map(&:size).max + SPACE_WIDTH }
-
-  number_of_row.times do |row_index|
-    number_of_col.times do |col_index|
-      # filenames_tableの行と列が逆で保存されているので、col_indexとrow_indexを入れ替えて出力させている
-      target_filename = filenames_table[col_index][row_index]
-      print target_filename&.ljust(widths[col_index])
+  # NOTE: file_names_tableの要素は行と列が出力したい形(縦並び)とは逆で保存されている
+  file_names_table = file_names.each_slice(row_count).to_a
+  widths = file_names_table.map { |col| col.map(&:size).max + SPACE_WIDTH }
+
+  row_count.times do |row_index|
+    col_count.times do |col_index|
+      # file_names_tableの行と列が逆で保存されているので、col_indexとrow_indexを入れ替えて出力させている
+      target_file_name = file_names_table[col_index][row_index]
+      print target_file_name&.ljust(widths[col_index])
     end
     print "\n"
   end
 end
 
-options, file_paths = search_option(ARGV)
-target_path = file_paths[0] || './'
-filenames = get_filenames(target_path, options)
-output(filenames)
+# directory_pathsには複数のpathを指定することは許容しているが、現時点でファイル名を表示するのは1番目に指定したディレクトリのみにしている。
+options, directory_paths = parse_options(ARGV)
+target_directory_path = directory_paths[0] || './'
```

Comment:
targetじゃないdirectory_pathがあるのか？と言ったらないと思うので、

```suggestion
directory_path = directory_paths[0] || './'
```

でもいいんじゃないかなーと思います。

Reply:
たしかにtargetじゃないものはないですね
修正しておきます！